codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -568,26 +568,29 @@ private ChannelFuture writeContinuationFrames(ChannelHandlerContext ctx, int str
         Http2Flags flags = new Http2Flags();
 
         if (headerBlock.isReadable()) {
-            // The frame header (and padding) only changes on the last frame, so allocate it once and re-use
-            int fragmentReadableBytes = min(headerBlock.readableBytes(), maxFrameSize);
-            ByteBuf buf = ctx.alloc().buffer(CONTINUATION_FRAME_HEADER_LENGTH);
-            writeFrameHeaderInternal(buf, fragmentReadableBytes, CONTINUATION, flags, streamId);
+
+            int fragmentReadableBytes;
+            ByteBuf buf = null;
 
             do {
                 fragmentReadableBytes = min(headerBlock.readableBytes(), maxFrameSize);
                 ByteBuf fragment = headerBlock.readRetainedSlice(fragmentReadableBytes);
 
                 if (headerBlock.isReadable()) {
+                    if (buf == null) {
+                        buf = ctx.alloc().buffer(CONTINUATION_FRAME_HEADER_LENGTH);
+                        writeFrameHeaderInternal(buf, fragmentReadableBytes, CONTINUATION, flags, streamId);
+                    }
                     ctx.write(buf.retainedSlice(), promiseAggregator.newPromise());
                 } else {
-                    // The frame header is different for the last frame, so re-allocate and release the old buffer
```

Comment:
Comment is still relevant and can be kept as-is

Reply:
> Comment is still relevant and can be kept as-is

Thanks for  your  reply，I have added it back.